src/CommunityToolkit.Maui/AppThemeObjectOfT.shared.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// Represents an object that is aware of the operating system theme.
+/// </summary>
+public class AppThemeObject<T>
+{
+	/// <summary>
+	/// The <see cref="object"/> that is used when the operating system uses light theme.
+	/// </summary>
+	public T? Light { get; set; }
+
+	/// <summary>
+	/// The <see cref="object"/> that is used when the operating system uses dark theme.
+	/// </summary>
+	public T? Dark { get; set; }
+
+	/// <summary>
+	/// The <see cref="object"/> that is used when the current theme is unspecified or
+	/// when a value is not provided for <see cref="Light"/> or <see cref="Dark"/>.
+	/// </summary>
+	public T? Default { get; set; }
+
+	internal BindingBase GetBinding()
+	{
+		var binding = new AppThemeBinding();
+
+		if (Light is not null)
```

Comment:
You would think so! However this is needed. If we don't do this it won't work correctly. For instance: when you only specify a color for Default and Dark theme and you set your device to Light theme, instead of Default, it will use Light which is then null.